assets/js/Components/App.js

```diff
@@ -154,14 +189,88 @@ class App extends React.Component {
 
   handleCourseRescan() {
     if (this.state.hasNewReport) {
-      this.setState({ hasNewReport: false, syncComplete: false })
+      this.setState({hasNewReport: false, 
+                     syncComplete: false, 
+                     contentUpdateCompleted: false,
+                     scanCompleted: false, 
+                     progress: 0})
       this.scanCourse()
         .then((response) => response.json())
-        .then(this.handleNewReport)
+        .then(() => this.getContentUpdate())
+        .then(() => this.pollBackgroundWorker())
+        .then(() => this.getReport())
     }
     this.forceUpdate()
   }
 
+  // As the worker process updates UDOIT with the latest content items, this method checks on whether or not
+  // This process has been completed
+  fetchUpdateData() {
+    fetch(`http://${window.location.hostname}:8000/udoit3/api/progress`)
```

Comment:
These can/should minimally be relative URL's. For example

`    fetch(`/udoit3/api/progress`)`

Otherwise they won't work on a hosted environment with SSL on regular ports. 

These might need to use the `BASE_URL` from the .env if this was changed from the default.